src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs

```diff
@@ -129,9 +129,10 @@ protected virtual unsafe IntPtr AppWndProc(IntPtr hWnd, uint msg, IntPtr wParam,
 
                 case WindowsMessage.WM_DPICHANGED:
                     {
-                        var dpi = ToInt32(wParam) & 0xffff;
+                        _dpi = (uint)wParam >> 16;
                         var newDisplayRect = Marshal.PtrToStructure<RECT>(lParam);
-                        _scaling = dpi / 96.0;
+                        _scaling = _dpi / 96.0;
+                        RefreshIcon();
```

Comment:
Question: is it safe to not call ClearIconCache on dpi changed? I imagine on each dpi change event app will create 2 bitmap handles per each icon. It potentially isn't a huge problem, but still leaves some unused resources taken by the app.

Reply:
Avalonia is per-monitor DPI aware, so this message is only relevant for the window's own icon display. Other processes (e.g. explorer, task manager) will continue to ask us for icons at whatever DPI they operate at.

Given this, clearing the cache would very likely increase memory usage, rather than decrease it.